retriever/engines/jsonengine.py

```diff
@@ -110,10 +110,14 @@ def insert_statement(self, values):
         else:
             newrows = values
         json_dumps = []
+        pretty = True if "pretty" in self.opts and self.opts["pretty"] is True else False
         for line_data in newrows:
             tuples = (zip(keys, line_data))
             write_data = OrderedDict(tuples)
-            json_dumps.append(json.dumps(write_data, ensure_ascii=False) + ",")
+            if not pretty:
```

Comment:
I think we can move the [check](https://github.com/weecology/retriever/pull/1287/files#diff-4b22ee1a8c7630feb86e76d515b9dcf9R113) here instead of` if not pretty` to 
`if "pretty" in self.opts and not self.opts["pretty"]":`

Reply:
It's inside a loop so it will have to check the condition for every row though it is not needed.
This is why I have written the check before the loop in a variable.

Reply:
Though the condition is not very complex we can move it down for readability of code. As you say.

Reply:
Thats fine, both will take constant time.